packages/playground/src/types.ts

```diff
@@ -537,3 +543,68 @@ export interface FunctionSchedulesFilter {
   createdTime?: Date;
   cronExpression?: string;
 }
+
+/* ************************
+   Annotations API types
+************************ */
+
+export type AnnotatedResourceType = 'file';
+export type AnnotationType =
+  | 'diagrams.AssetLink'
+  | 'diagrams.FileLink'
+  | 'diagrams.PipeSymbol'
+  | 'documents.ExtractedText';
+export type LinkedResourceType = 'file' | 'asset';
+export type AnnotationStatus = 'suggested' | 'approved' | 'rejected';
+
+export interface AnnotationModel extends AnnotationCreate {
+  id: CogniteInternalId;
+  createdTime: Date;
+  lastUpdatedTime: Date;
+}
+
+export interface AnnotationCreate {
+  annotatedResourceType: AnnotatedResourceType;
+  annotatedResourceId?: CogniteInternalId;
+  annotatedResourceExternalId?: CogniteExternalId;
+  annotationType: AnnotationType;
+  creatingApp: string;
+  creatingAppVersion: string;
+  creatingUser: string | null;
+  data: object;
+  linkedResourceType?: LinkedResourceType;
+  linkedResourceId?: CogniteInternalId;
+  linkedResourceExternalId?: CogniteExternalId;
+  status: AnnotationStatus;
+}
+
+export interface AnnotationChangeById extends InternalId, AnnotationUpdate {}
+
+export interface AnnotationUpdate {
+  update: {
+    annotationType?: SinglePatchRequiredString;
+    data?: SetField<object>;
+    linkedResourceType?: SinglePatchString;
+    linkedResourceId?: SinglePatch<CogniteInternalId>;
+    linkedResourceExternalId?: SinglePatch<CogniteExternalId>;
+  };
+}
+
+export interface AnnotationFilterRequest
+  extends AnnotationFilter,
+    FilterQuery {}
+
+export interface AnnotationFilter {
+  filter: AnnotationFilterProps;
+}
+
+export interface AnnotationFilterProps {
+  annotatedResourceType: AnnotatedResourceType;
+  annotatedResourceIds: IdEither[];
+  annotationType?: AnnotationType;
+  creatingApp?: string;
+  creatingUser?: string | null;
```

Comment:
what does null do here?

Reply:
`null` is a special value that signals that the annotation was not created by a particular user, i.e. the annotation was created by a service.